runtime/lua/vim/lsp.lua

```diff
@@ -607,6 +608,20 @@ do
         })
       end
     end
+
+    vim.schedule(function()
+      nvim_exec_autocmds('LspRequest', {
```

Comment:
That's definitely contrary to the semantics of notifications. Should at least be documented.

Also should call out `data.request.method` in the docs, if it isn't already, especially the docs related to this PR.

Reply:
If notifications currently don't emit a `LspRequest` event I'd be inclined to introduce a new `LspNotify` or `LspNotification` autocmd. Requests and notifications are semantically different in the protocol and I guess in how the events behave as well. Notifications don't have the state transition because you cannot cancel a notification, so by adding a new trigger here we'd invalidate part of the documented `LspRequest` behavior. 

That would then happen in https://github.com/neovim/neovim/blob/f771d6247147b393238fe57065a96fb5e9635358/runtime/lua/vim/lsp.lua#L1572-L1577


@justinmk is that okay with you?

Reply:
Seems unavoidable :/